### PR TITLE
DAOS-6389 test: datamover dsync tests

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag-vm: pr,-hw datamover,-hw
Test-tag-hw-small: pr,hw,small datamover,hw,small
Test-tag-hw-large: pr,hw,large datamover,hw,large

Added DsyncCommand and Dsync to the data mover
utils and test base.

Added dsync variant to dm_posix_types.

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/data_mover_test_base.py [hunk 1]
```diff
@@ -407,6 +441,76 @@ def set_dcp_params(self,
...
+                    self.dsync_cmd.set_dsync_params(
+                        prefix=src_cont.path.value,
+                        src=src_cont.path.value + src_path)
+
+        # Set the destination params
+        if dst_type == "POSIX":
+            self.dsync_cmd.set_dsync_params(
+                dst=dst_path)
+        elif dst_type == "DAOS_UUID":
+            param = param = self._format_daos_path(dst_pool, dst_cont, dst_path)
```
Comment (shimizukko) [inline src/tests/ftest/util/data_mover_test_base.py:501]:
```suggestion
            param = self._format_daos_path(dst_pool, dst_cont, dst_path)
```
typo?

**Review by shimizukko — APPROVED** [review]
LGTM

File: src/tests/ftest/util/data_mover_test_base.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/data_mover_test_base.py]: Yes, fixed :)

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by dsikich — APPROVED** [review]

**Pull request merged**